### PR TITLE
feat(tag): add filter tag variant

### PR DESCRIPTION
**Pull request by cal-smith**
The [filtering patterns](https://next.carbondesignsystem.com/patterns/filtering) ([v9 link](https://www.carbondesignsystem.com/patterns/filtering/)) show closable tags used outside of multi-select dropdowns. Right now the only "filter" or "closable" tag is implemented as `.#{$prefix}--list-box__selection--multi`.

I've done up an implementation of this as `.#{$prefix}--tag--filter`, it follows `.#{$prefix}--list-box__selection--multi` semantics, but using the existing `.#{$prefix}--tag` base class.

v9
image

v10
image

### CODE REVIEW

**Comment by joshblack** [comment]
cc @IBM/carbon-designers are we good with adding this in? 👍

**Review by aagonzales — COMMENTED** [review]
Yes, this is a great addition! Only change is in the V10 version, the `x` icon should use the 16x16 size and not 20x20

File: src/components/tag/_tag.scss [hunk 1]
```diff
@@ -139,6 +158,30 @@
...
+    @include tag-theme--x($interactive-02, $inverse-01);
+    cursor: pointer;
+    padding-right: rem(2px); // Align with hover circle of X button
+  }
+
+  .#{$prefix}--tag--filter > svg {
+    fill: $inverse-01;
+    margin-left: rem(4px);
+    width: rem(20px);
+    height: rem(20px);
```
Comment (jnm2377) [inline src/components/tag/_tag.scss:173]: Regarding Anna's comment, everything looks good if we just remove these so it's 16 x 16

**Review by jnm2377 — APPROVED** [review]

File: src/components/tag/_tag.scss (hunk 1 above)
Comment (cal-smith) [reply src/components/tag/_tag.scss:173]: Cool sounds good 👍

**Comment by cal-smith** [comment]
Fixed 👍

**Comment by aagonzales** [comment]
Opps sorry I should have checked the spec, but can we now add 4px of padding to the right of the x and then make the hover circle be 20x20px?

image

**Comment by cal-smith** [comment]
For sure

**Comment by cal-smith** [comment]
Something like this @aagonzales?

image

I set it to 20x20, but padded the icon down by 4px so it _should_ be 16x16 still

**Comment by aagonzales** [comment]
Yes, but did the x get smaller? Kind of looks like the 4px padding made the x smaller but the placement and hover size looks correct just the icon size looks wrong now.

**Comment by cal-smith** [comment]
Hmm let me play with it a bit ... worst case I'll fall back to pseudo elements for the hover state

**Comment by cal-smith** [comment]
There we go ... the icon path is 8.016x8.016 as 16x16, bumping it up to 20x20 makes it 10x10 (roughly) and padding it down by 2px gives us 8.016 again 🙌

**Review by aagonzales — APPROVED** [review]
Yes, that looks good! Thanks for the updates, sorry for the run-around.

**Pull request merged**